src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -260,14 +251,22 @@ void FFN<OutputLayerType, InitializationRuleType>::Gradient(
     gradient.zeros();
   }
 
-  Evaluate(parameters, i, false);
+  Evaluate(parameters, begin, batchSize, false);
 
-  outputLayer.Backward(std::move(boost::apply_visitor(outputParameterVisitor,
-      network.back())), std::move(currentTarget), std::move(error));
+  outputLayer.Backward(
+      std::move(boost::apply_visitor(outputParameterVisitor, network.back())),
+      std::move(responses.cols(begin, begin + batchSize - 1)),
+      std::move(error));
 
   Backward();
   ResetGradients(gradient);
-  Gradient();
+  Gradient(std::move(predictors.cols(begin, begin + batchSize - 1)));
```

Comment:
I missed ``gradient /= batchSize;``, do you like to add that step or should I commit the changes to my branch?

Reply:
@zoq Done!

Reply:
Sorry for the slow response here.  If we normalize by `batchSize`, then to take the same amount of a step when I have `batchSize = 2` in a single iteration of SGD, I have to double the step size.  So I think we should not normalize, what do you think?

Reply:
Fine for me, in the future we should write a tutorial to clear things like that.